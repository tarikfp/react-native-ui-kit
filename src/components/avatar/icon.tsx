import PropTypes from "prop-types";
import React from "react";
import { StyleProp, StyleSheet, View, ViewStyle } from "react-native";
import { StylePropType } from "../utils";
import Icon, { IconProps } from "../vector-icons";

const DEFAULT_ICON_SIZE = 32;

type Props = {
  /**
   * size of the avatar
   * @default 32
   *
   */
  size?: number;
  /**
   * background color of the avatar
   * @default rgba(35, 47, 72, 1)
   */
  backgroundColor?: string;
  /**
   * avatar style
   */
  style?: StyleProp<ViewStyle>;
  /**
   * icon name
   */
  name: IconProps["name"];
  /**
   * icon type
   */
  type: IconProps["type"];
  /**
   * icon color
   */
  color?: IconProps["color"];
  /**
   * icon props
   */
  iconProps?: IconProps;
};

function AvatarIcon({
  size = DEFAULT_ICON_SIZE,
  backgroundColor = "rgba(35, 47, 72, 1)",
  iconProps,
  name,
  color,
  type,
  style
}: Props) {
  return (
    <View
      style={[
        { width: size, height: size, borderRadius: size / 2, backgroundColor },
        styles.wrapper,
        style
      ]}>
      <Icon
        size={size * 0.7}
        name={name}
        type={type}
        color={color}
        {...iconProps}
      />
    </View>
  );
}

export default AvatarIcon;

AvatarIcon.propTypes = {
  size: PropTypes.number,
  backgroundColor: PropTypes.string,
  style: StylePropType,
  name: PropTypes.string,
  type: PropTypes.string,
  color: PropTypes.string,
  iconProps: PropTypes.object
};

AvatarIcon.displayName = "Avatar.Icon";

const styles = StyleSheet.create({
  wrapper: {
    justifyContent: "center",
    alignItems: "center"
  }
});
