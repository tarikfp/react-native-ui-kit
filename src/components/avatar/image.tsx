import PropTypes from "prop-types";
import * as React from "react";
import {
  Image,
  ImageProps,
  ImageSourcePropType,
  ImageStyle,
  StyleProp,
  View,
  ViewStyle,
} from "react-native";
import { StylePropType } from "../utils";

const DEFAULT_AVATAR_IMAGE_SIZE = 32;

type Props = ImageProps & {
  /**
   * Image to display for the `Avatar`.
   */
  source: ImageSourcePropType;
  /**
   * Size of the avatar.
   */
  size?: number;
  /**
   * style of the image wrapper
   */
  style?: StyleProp<ViewStyle>;
  /**
   * style of the image
   */
  imageStyle?: StyleProp<ImageStyle>;
  /**
   * background color of the image wrapper
   */
  backgroundColor?: string;
};

function AvatarImage({
  source,
  size = DEFAULT_AVATAR_IMAGE_SIZE,
  style,
  imageStyle,
  backgroundColor,
  ...imageProps
}: Props) {
  return (
    <View
      style={[
        {
          width: size,
          height: size,
          borderRadius: size / 2,
          backgroundColor,
        },
        style,
      ]}>
      <Image
        source={source}
        style={[
          { width: size, height: size, borderRadius: size / 2 },
          imageStyle,
        ]}
        {...imageProps}
      />
    </View>
  );
}

export default AvatarImage;

AvatarImage.propTypes = {
  source: PropTypes.object,
  size: PropTypes.number,
  style: StylePropType,
  imageStyle: StylePropType,
  backgroundColor: PropTypes.string,
};

AvatarImage.displayName = "Avatar.Image";
