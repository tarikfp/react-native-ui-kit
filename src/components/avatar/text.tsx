import PropTypes from "prop-types";
import React from "react";
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  View,
  ViewStyle,
} from "react-native";
import { StylePropType } from "../utils";

const DEFAULT_AVATAR_TEXT_SIZE = 32;

type Props = {
  /**
   * text to display on avatar
   */
  label: string;
  /**
   * size of the avatar
   * @default 32
   *
   */
  size?: number;
  /**
   * label color
   * @default #FFFFFF
   */
  labelColor?: string;
  /**
   * background color of the avatar
   * @default rgba(35, 47, 72, 1)
   */
  backgroundColor?: string;
  /**
   * avatar label wrapper style
   */
  style?: StyleProp<ViewStyle>;
  /**
   * avatar label style
   */
  labelStyle?: StyleProp<TextStyle>;
};

function AvatarText({
  label,
  size = DEFAULT_AVATAR_TEXT_SIZE,
  labelColor = "#FFFFFF",
  backgroundColor = "rgba(35, 47, 72, 1)",
  labelStyle,
  style,
}: Props) {
  return (
    <View
      style={[
        { width: size, height: size, borderRadius: size / 2, backgroundColor },
        styles.wrapper,
        style,
      ]}>
      <Text
        style={[
          styles.text,
          {
            color: labelColor,
            fontSize: size / 2.25,
            lineHeight: size,
          },
          labelStyle,
        ]}
        numberOfLines={1}>
        {label}
      </Text>
    </View>
  );
}

export default AvatarText;

AvatarText.propTypes = {
  label: PropTypes.string,
  size: PropTypes.number,
  labelColor: PropTypes.string,
  backgroundColor: PropTypes.string,
  style: StylePropType,
  labelStyle: StylePropType,
};

AvatarText.displayName = "Avatar.Text";

const styles = StyleSheet.create({
  wrapper: {
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    textAlign: "center",
    textAlignVertical: "center",
  },
});
