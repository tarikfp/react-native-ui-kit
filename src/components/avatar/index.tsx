import { default as Icon } from "./icon";
import { default as Image } from "./image";
import { default as Text } from "./text";

export default {
  Icon,
  Image,
  Text
};
