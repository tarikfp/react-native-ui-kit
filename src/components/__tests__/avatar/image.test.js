import * as React from "react";
import renderer from "react-test-renderer";
import { default as Avatar } from "../../avatar";

describe("tests avatar with image component", () => {
  it("renders avatar with image with source uri and custom style", () => {
    const tree = renderer
      .create(
        <Avatar.Image
          style={{ width: 50, height: 50 }}
          source={{
            uri: "https://reactjs.org/logo-og.png",
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders avatar with image with custom size", () => {
    const tree = renderer
      .create(
        <Avatar.Image
          size={70}
          style={{ width: 50, height: 50, borderRadius: 8 }}
          source={{
            uri: "https://reactjs.org/logo-og.png",
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders avatar with image with custom image style", () => {
    const tree = renderer
      .create(
        <Avatar.Image
          style={{ width: 50, height: 50, borderRadius: 8 }}
          source={{
            uri: "https://reactjs.org/logo-og.png",
          }}
          imageStyle={{ width: 30, height: 30 }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders avatar with image with backgroundColor", () => {
    const tree = renderer
      .create(
        <Avatar.Image
          style={{ width: 50, height: 50, borderRadius: 8 }}
          source={{
            uri: "https://reactjs.org/logo-og.png",
          }}
          backgroundColor="royalblue"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
