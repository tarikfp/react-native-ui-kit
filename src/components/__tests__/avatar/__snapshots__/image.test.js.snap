// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests avatar with image component renders avatar with image with backgroundColor 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "royalblue",
        "borderRadius": 16,
        "height": 32,
        "width": 32,
      },
      Object {
        "borderRadius": 8,
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactjs.org/logo-og.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 16,
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`tests avatar with image component renders avatar with image with custom image style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderRadius": 16,
        "height": 32,
        "width": 32,
      },
      Object {
        "borderRadius": 8,
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactjs.org/logo-og.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 16,
          "height": 32,
          "width": 32,
        },
        Object {
          "height": 30,
          "width": 30,
        },
      ]
    }
  />
</View>
`;

exports[`tests avatar with image component renders avatar with image with custom size 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderRadius": 35,
        "height": 70,
        "width": 70,
      },
      Object {
        "borderRadius": 8,
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactjs.org/logo-og.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "width": 70,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`tests avatar with image component renders avatar with image with source uri and custom style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": undefined,
        "borderRadius": 16,
        "height": 32,
        "width": 32,
      },
      Object {
        "height": 50,
        "width": 50,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://reactjs.org/logo-og.png",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 16,
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
  />
</View>
`;
